api/src/db/models/user_models.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+
+from sqlalchemy import ForeignKey
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.adapters.db.type_decorators.postgres_type_decorators import LookupColumn
+from src.db.models.base import ApiSchemaTable, TimestampMixin
+from src.db.models.lookup_models import LkExternalUserType
+
+
+class User(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+
+
+class LinkExternalUser(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "link_external_user"
+
+    link_external_user_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+
+    external_user_id: Mapped[str]
+
+    user_id: Mapped[uuid.UUID] = mapped_column(UUID, ForeignKey(User.user_id), index=True)
+    user: Mapped[User] = relationship(User)
+
+    external_user_type: Mapped[int] = mapped_column(
+        "external_user_type_id",
+        LookupColumn(LkExternalUserType),
+        ForeignKey(LkExternalUserType.external_user_type_id),
+        index=True,
+    )
+
+    email: Mapped[str]
+
+    first_name: Mapped[str]
+
+    last_name: Mapped[str]
```

Comment:
NOTE: regardless of anything else, we should hold on merging this until I've gotten more clarification. We might not have first/last name as those require ID proofing and I don't know if we intend for users to be ID proofed 100% of the time.

Reply:
Converted this PR to a Draft PR so we don't accidentally merge